File: python/remix-fa23/solidworks.py
def solve(sw, h, w, d):
    new = [[' ' for i in range(w + d + 1)] for x in range(h + d + 1)]
    edges = []
    

    # transpose the original shape on the new list
    for row in range(h):
        for col in range(w):
            new[row + d + 1][col + d + 1] = sw[row][col]

            if sw[row][col] == '+':
                edges.append((row + d + 1, col + d + 1))
    
    # extend edges
    for edge in edges:
        for i in range(1, d + 1):
            if new[edge[0] - i][edge[1] - i] not in ('-', '|', '+'):
                new[edge[0] - i][edge[1] - i] = '\\'
    
    # place back side of shape
    for row in range(h):
        for col in range(w):
            if new[row][col] == ' ':   
                new[row][col] = sw[row][col]            
    
    return new


def main():
    for _ in range(int(input())):
        h, w, d = map(int, input().split())
        sw = []
        
        for _ in range(h):
            sw.append(list(input()))
        
        sol = solve(sw, h, w, d)
        print("\n".join(["".join(x) for x in sol]))    

main()

File: python/remix-fa23/solidworks.cpp
    #include <iostream>
    #include <vector>
    #include <algorithm>

    using namespace std;

    void solve(vector<vector<char>> sw, int h, int w, int d) {
        vector<char> new_row(w + d + 1, ' ');
        vector<vector<char>> ns((h + d + 1));
        fill(ns.begin(), ns.end(), new_row);

        vector<pair<int, int>> edges(0);

        // transpose the original shape on the new list
        for (int row = 0; row < h; row++) {
            for (int col = 0; col < w; col++) {
                ns[row + d + 1][col + d + 1] = sw[row][col];

                if (sw[row][col] == '+') {
                    edges.push_back(make_pair(row +d + 1, col + d + 1));
                }
            }
        }

        // Extend edges
        for (int x = 0; x < edges.size(); x++) {
            for (int i = 1; i < d + 1; i++) {
                pair<int, int> p = edges[x];
                char ch = ns[p.first - i][p.second - i];
                
                if (ch == '-' || ch == '|' || ch == '+') {
                    continue;
                }

                ns[p.first - i][p.second - i] = '\\';
            }
        }

        // place backside of shape
        for (int row = 0; row < h; row++) {
            for (int col = 0; col < w; col++) {
                if (ns[row][col] == ' ') {
                    ns[row][col] = sw[row][col];
                }
            }
        }


        // print the vectors
        for (vector<char> v: ns) {
            for (char c: v) {
                cout << c;
            }
            cout << "\n";
        }
    }

    int main() {
        int T;
        cin >> T;

        for (int i = 0; i < T; i++) {
            int h, w, d;
            cin >> h >> w >> d;

            vector<vector<char>> sw(h);

            cin.ignore(); // idk why this is here

            for (int j = 0; j < h; j++) {
                string temp;
                getline(cin, temp);

                vector<char> r(temp.begin(), temp.end());
                sw[j] = r;
            }

            solve(sw, h, w, d);
        }
    }